torch/csrc/jit/codegen/cuda/transform_view.cpp

```diff
@@ -510,7 +510,8 @@ class AnalyzeViewTransformation {
       }
 
       if ((new_view_index == new_view_.size() ||
-           (new_view_[new_view_index + 1] != 1)) &&
+           (new_view_.size() > new_view_index + 1 &&
+            new_view_[new_view_index + 1] != 1)) &&
```

Comment:
This actually looks about right to me as well. This case should be handled later at
line 566:
`if (current_size == 1 && isImplicitBroadcast(original_view_index)) {`

Any chance you have a repro of the failing case?

Reply:
Repro: `FusionViewAllShmoo_CUDA` test; I think `new_view_[new_view_index + 1]` (above) is the out-of-bounds access. Internally we have a build mode that catches these out of bound accesses, you can probably add an assert for the access to do the same check

Reply:
hmmm... There's an actual cpp tests that's failing?! Let me double check that.

@naoya has been advocating us abandoning `std::vector::operator[]` :)